shells/default/default.nix

```diff
@@ -0,0 +1,78 @@
+{
+  lib,
+  mkShell,
+  just,
+  hivemind,
+  commitizen,
+  alejandra,
+  pre-commit-check,
+  flox,
+  flox-pkgdb,
+  flox-env-builder,
+  flox-tests,
+  flox-tests-end2end,
+  flox-env-builder-tests,
+  flox-pkgdb-tests,
+  ci ? false,
+}: let
+  # For use in GitHub Actions and local development.
+  ciPackages = [
+    (flox-pkgdb-tests.override {
+      PROJECT_TESTS_DIR = "/pkgdb/tests";
+      PKGDB_BIN = null;
+      PKGDB_IS_SQLITE3_BIN = null;
+      PKGDB_SEARCH_PARAMS_BIN = null;
+    })
+    (flox-env-builder-tests.override {
+      PROJECT_TESTS_DIR = "/env-builder/tests";
+      PKGDB_BIN = null;
+      ENV_BUILDER_BIN = null;
+    })
+    (flox-tests.override {
+      PROJECT_TESTS_DIR = "/tests";
+      PKGDB_BIN = null;
+      ENV_BUILDER_BIN = null;
+      FLOX_BIN = null;
+    })
+    (flox-tests.override {
+      PROJECT_NAME = "flox-tests-end2end";
+      PROJECT_TESTS_DIR = "/tests";
+      PKGDB_BIN = null;
+      ENV_BUILDER_BIN = null;
+      FLOX_BIN = null;
+    })
+  ];
+
+  devPackages =
+    flox-pkgdb.devPackages
+    ++ flox-env-builder.devPackages
+    ++ flox.devPackages
+    ++ [
+      just
+      hivemind
+      commitizen
+      alejandra
+    ];
+in
+  mkShell (
+    {
+      name = "flox-dev";
+
+      inputsFrom = [
+        flox-pkgdb
+        flox-env-builder
+        flox
```

Comment:
doesn't `flox` have `flox-pkgdb` and `flox-env-builder` as dependencies?
should this rather be something along this?

```
[
	flox-pkgdb
    flox-env-builder
    (flox.override {
		flox-pkgdb = null;
		flox-env-builder = null;
	})
]
```

Reply:
Yes, but in this case (development environment) we want to use development versions of those projects. Setting things to null will use things from path.

In CI we actually need the opposite.

Reply:
It's just, you dont want to rebuild `flox-pkgdb` when pkgdb files changed and you ope the dev shell?

In CI i get you want to use the built ones for testing `cargo build flox`